fix: implement signalfd

yoinspiration opened the pull request:
- Add Signalfd file descriptor implementation (api/src/file/signalfd.rs)
- Implement signalfd4 system call handler (api/src/syscall/fs/signalfd.rs)
- Register signalfd4 in syscall router (api/src/syscall/mod.rs)
- Export signalfd module in file/mod.rs and syscall/fs/mod.rs

This fixes the panic when reading from signalfd4 file descriptor. Previously, signalfd4 returned a DummyFd which would panic on read. Now it returns a proper Signalfd that can read signals correctly.

Fixes #15

AsakuraMizu commented:
Please refer to https://docs.github.com/zh/get-started/git-basics/configuring-git-to-handle-line-endings and update your commit. I can not identify your changes in diff.

yoinspiration commented:
I've updated the commit according to the GitHub documentation on configuring Git to handle line endings.

## What I've Done

1. Added `.gitattributes` file to ensure consistent line endings (LF)
2. Reorganized the commit to clearly show only the actual code changes

## Result

The commit now displays clean, readable diffs:
- **6 files changed, 300 insertions(+), 3 deletions(-)**
- New files: `api/src/file/signalfd.rs` (194 lines) and `api/src/syscall/fs/signalfd.rs` (65 lines)
- Modified files show only relevant changes (e.g., `api/src/file/mod.rs` shows just 1 line: `+pub mod signalfd;`)

The diff statistics are consistent with or without whitespace ignored, confirming the line ending issue is resolved. You should now be able to clearly identify all code changes in the diff.

yoinspiration commented:
Thank you for the thorough review! I've addressed all the comments:

1. **SFD flag constants**: Updated to use `O_CLOEXEC` and `O_NONBLOCK` from `linux_raw_sys::general` instead of hardcoded values, following the same pattern as `pipe.rs` and `event.rs`.

2. **Chinese comments**: Translated all Chinese comments to English for consistency with the codebase.

3. **Unsafe block formatting**: Removed the empty line and updated the comment to use singular "field" since only one field is accessed.

All changes have been committed and pushed. The implementation now correctly handles the signalfd4 flags and maintains consistency with the rest of the codebase.

yoinspiration commented on api/src/file/signalfd.rs:
```diff
@@ -0,0 +1,192 @@
+                    let sfd_info = SignalfdSiginfo::from_signal_info(&sig_info);
+                    
+                    // Write the structure to the destination buffer
+                    let bytes = unsafe {
```
**Fixed** ✅

Replaced the unsafe transmute with `zerocopy::IntoBytes` for safe C struct-to-bytes conversion.

**Changes made:**

1. Added `use zerocopy::IntoBytes;` import (line 16)
2. Added `#[derive(IntoBytes)]` to the `SignalfdSiginfo` struct (line 26)
3. Replaced the unsafe code block with `sfd_info.as_bytes()` (line 134)

**Relevant code:**

```rust
// Line 26-27: Added IntoBytes derive
#[repr(C)]
#[derive(IntoBytes)]
struct SignalfdSiginfo {
    // ...
}
```

```rust
// Line 133-135: Replaced unsafe transmute with safe conversion
// Write the structure to the destination buffer
let bytes = sfd_info.as_bytes();
dst.write(bytes)?;
```

The code now safely converts the C struct to bytes using `zerocopy` without any unsafe blocks, which is consistent with the approach used in `api/src/vfs/dev/event.rs`.

yoinspiration commented:
True, my bad — thanks for updating it!
